stdlib/list.ml

```diff
@@ -220,6 +220,16 @@ let rec remove_assq x = function
   | [] -> []
   | (a, _ as pair) :: l -> if a == x then l else pair :: remove_assq x l
 
+let rec fold_assoc_left f accu l =
```

Comment:
The question is maybe naïve but why not writting `fold_assoc_left/right` using fold_left/right?

```ocaml
let fold_assoc_left f accu l = 
    fold_left (fun accu (k, v) -> f accu k v) accu l 
```

Reply:
Having an inlined definition is expected to be slightly more efficient. In a stdlib, I think we can trade a little redundancy in exchange for efficiency, when the duplication is not too bad maintenance-wise. For `fold_assoc_left` I redefined as the code is simple and natural, but notice that for `Hashtbl.fold_pair` I did use the approach you mention, because the code for folding over a hashtable is sensibly more delicate, so I didn't want to duplicate it.

Reply:
Thanks for the reply!